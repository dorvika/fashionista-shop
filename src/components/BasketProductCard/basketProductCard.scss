.cart__product {
  display: flex;
  justify-content: space-between;
  height: 250px;
  margin: 20px 0;
  box-shadow: 5px 5px 20px -1px rgba(0, 0, 0, 0.3);

  &__info,
  &__counter,
  &__sum {
    align-self: center;
    font-weight: 200;
    margin-left: 40px;
  }

  &__sum {
    font-size: 20px;
  }

  &__title {
    font-weight: 200;
  }

  &__delete {
    align-self: flex-start;
    padding: 0 10px;
  }

  &__info {
    font-size: 16px;
    flex-grow: 1;
  }

  &__counter {
    display: flex;
    align-items: center;
    button {
      border: none;
      background-color: white;
      font-size: 35px;
      cursor: pointer;
      font-weight: 200;
      transition: all 0.2s ease-in;
      &:hover {
        font-weight: 600;
      }
    }
    span {
      font-size: 20px;
      padding: 0 20px;
    }
  }
  &__delete {
    border: none;
    background-color: white;
    font-size: 35px;
    cursor: pointer;
    font-weight: 200;
    transition: all 0.2s ease-in;
    &:hover {
      font-weight: 600;
    }
  }
}

@media screen and (max-width: 1100px) {
  .cart__product {
    &__sum {
      font-size: 18px;
    }
    &__counter {
      button {
        font-size: 30px;
      }
      span {
        font-size: 18px;
      }
    }
  }
}

@media screen and (max-width: 1025px) {
  .cart__product {
    &__info,
    &__counter {
      margin-left: 10px;
    }
  }
}

@media screen and (max-width: 960px) {
  .cart__product {
    &__sum {
      margin-left: 20px;
    }
  }
}

@media screen and (max-width: 940px) {
  .cart__product {
    &__sum {
      width: 10%;
    }
    &__counter {
      margin-right: 30px;
    }
  }
}

@media screen and (max-width: 900px) {
  .cart__product {
    &__info {
      max-width: 250px;
    }
  }
}

@media screen and (max-width: 768px) {
  .cart__product {
    &__title {
      font-size: 17px;
    }
    &__sum {
      font-size: 16px;
    }
    &__counter {
      button {
        font-size: 25px;
      }
      span {
        font-size: 16px;
      }
    }
  }
}

@media screen and (max-width: 720px) {
  .cart__product {
    position: relative;
    flex-direction: column;
    align-items: center;
    margin: 20px auto;
    height: auto;
    width: 50%;
    img {
      margin-top: 20px;
    }

    &__info {
      max-width: 50%;
      text-align: center;
      margin-left: 0;
      h3 {
        margin-top: 10px;
      }
      p {
        margin: 10px 0;

        &:nth-child(3) {
          display: none;
        }
      }
    }

    &__counter {
      margin: 0 0 10px 0;
    }

    &__sum {
      display: none;
    }

    &__delete {
      position: absolute;
      right: 0;
    }
  }
}

@media screen and (max-width: 650px) {
  .cart__product {
    width: 70%;
  }
}

@media screen and (max-width: 500px) {
  .cart__product {
    width: 100%;
  }
}

@media screen and (max-width: 410px) {
  .cart__product {
    &__info {
      max-width: 80%;
    }
  }
}
