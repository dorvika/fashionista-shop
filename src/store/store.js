import {
  legacy_createStore as createStore,
  combineReducers,
  compose,
  applyMiddleware,
} from "redux";
import thunk from "redux-thunk";
import productReducer from "./products/reducer";
import favoriteReducer from "./favorities/reducer";
import basketReducer from "./basket/reducer";
import modalReducer from "./modal/reducer";
import { setProductsToLS } from "../helpers/localStorage";

export const rootReducer = combineReducers({
  products: productReducer,
  favorites: favoriteReducer,
  basket: basketReducer,
  modal: modalReducer,
});

const devTools = window.__REDUX_DEVTOOLS_EXTENSION__
  ? window.__REDUX_DEVTOOLS_EXTENSION__()
  : (f) => f;

const syncWithLS = (store) => (next) => (action) => {
  const result = next(action);
  if (action.type === "TOGGLE_FAVORITE") {
    setProductsToLS("Products in Favorite", store.getState().favorites);
    return result;
  }
  if (action.type === "ADD_TO_BASKET" || action.type === "DELETE_FROM_BASKET") {
    setProductsToLS("Products in Basket", store.getState().basket);
    return result;
  }
  return next(action);
};

const store = createStore(
  rootReducer,
  compose(applyMiddleware(thunk, syncWithLS), devTools)
);

export default store;
