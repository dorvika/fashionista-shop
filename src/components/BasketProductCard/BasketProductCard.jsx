import "./basketProductCard.scss";
import { useDispatch } from "react-redux";
import { addToBasket, deleteFromBasket } from "../../store/basket/actions";
import { openModal } from "../../store/modal/actions";
import Button from "../Button";

export default function BasketProductCard({ productObj }) {
  const dispatch = useDispatch();
  const { name, price, img, color, qty } = productObj;

  const onAdd = () => {
    dispatch(addToBasket(productObj));
  };

  return (
    <div className="cart__product">
      <img src={img} width={200} height={250} alt={name} />
      <div className="cart__product__info">
        <h3 className="cart__product__title">{name}</h3>
        <p>{price} грн</p>
        <p>Колір: {color}</p>
      </div>
      <div className="cart__product__counter">
        <Button
          onClick={
            qty === 1
              ? () => {
                  dispatch(
                    openModal({
                      content: "Ви бажаєте видалити даний товар із кошика?",
                      action: "delete",
                      product: productObj,
                    })
                  );
                }
              : () => dispatch(deleteFromBasket(productObj.sku))
          }
        >
          -
        </Button>
        <span>{qty}</span>
        <Button onClick={onAdd}>+</Button>
      </div>
      <div className="cart__product__sum"> {qty * price} грн</div>
      <Button
        className="cart__product__delete"
        onClick={() =>
          dispatch(
            openModal({
              content: "Ви бажаєте видалити даний товар із кошика?",
              action: "delete",
              product: productObj,
              deleteTotallyFlag: true,
            })
          )
        }
      >
        &times;
      </Button>
    </div>
  );
}
