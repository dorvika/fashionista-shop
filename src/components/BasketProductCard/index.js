import BasketProductCard from "./BasketProductCard";

export default BasketProductCard;
