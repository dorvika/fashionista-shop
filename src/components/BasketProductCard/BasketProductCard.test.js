import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import BasketProductCard from "./BasketProductCard";
import { rootReducer } from "../../store/store";
import { legacy_createStore as createStore } from "redux";

describe("<Button /> component", () => {
  const store = createStore(rootReducer);
  it("should match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <BasketProductCard
          productObj={{
            name: "Dress",
            price: "699",
            img: null,
            color: "white",
            qty: 3,
          }}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
