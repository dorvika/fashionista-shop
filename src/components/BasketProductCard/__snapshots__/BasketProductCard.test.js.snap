// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component should match snapshot 1`] = `
<div>
  <div
    class="cart__product"
  >
    <img
      alt="Dress"
      height="250"
      width="200"
    />
    <div
      class="cart__product__info"
    >
      <h3
        class="cart__product__title"
      >
        Dress
      </h3>
      <p>
        699
         грн
      </p>
      <p>
        Колір: 
        white
      </p>
    </div>
    <div
      class="cart__product__counter"
    >
      <button
        type="button"
      >
        -
      </button>
      <span>
        3
      </span>
      <button
        type="button"
      >
        +
      </button>
    </div>
    <div
      class="cart__product__sum"
    >
       
      2097
       грн
    </div>
    <button
      class="cart__product__delete"
      type="button"
    >
      ×
    </button>
  </div>
</div>
`;

exports[`<Button /> component should render 1`] = `
<div>
  <div
    class="cart__product"
  >
    <img
      alt="Dress"
      height="250"
      width="200"
    />
    <div
      class="cart__product__info"
    >
      <h3
        class="cart__product__title"
      >
        Dress
      </h3>
      <p>
        699
         грн
      </p>
      <p>
        Колір: 
        white
      </p>
    </div>
    <div
      class="cart__product__counter"
    >
      <button
        type="button"
      >
        -
      </button>
      <span>
        3
      </span>
      <button
        type="button"
      >
        +
      </button>
    </div>
    <div
      class="cart__product__sum"
    >
       
      2097
       грн
    </div>
    <button
      class="cart__product__delete"
      type="button"
    >
      ×
    </button>
  </div>
</div>
`;
